crates/sui-core/src/authority/authority_per_epoch_store.rs

```diff
@@ -2152,6 +2367,15 @@ impl AuthorityPerEpochStore {
                     return Ok(ConsensusCertificateResult::Ignored);
                 }
 
+                if let Some(deferral_key) = self.should_defer(&certificate) {
```

Comment:
This is pretty neat. In the case of congestion control, if a txn needs to be deferred multiple times, I guess we just defer it until next round, and then when the next round comes we can choose to be agnostic (or not) about the fact that this txn has already been deferred before and defer it one more time.

Reply:
We will eventually need to some kind of starvation prevention - certificates should not stay in the queue forever.

But since we are serializing the entire consensus tx, we will have access to the round that it was originally committed in, so we can use that to detect starvation